packages/app/src/background/services/credentials/utils.ts

```diff
@@ -0,0 +1,243 @@
+    verifiableCredential: serializeVerifiableCredential(cryptkeeperVerifiableCredential.verifiableCredential),
+    metadata: cryptkeeperVerifiableCredential.metadata,
+  });
+}
+
+/**
+ * Deserializes a CryptkeeperVerifiableCredential JSON string into a CryptkeeperVerifiableCredential object.
+ * @param serializedCryptkeeperVerifiableCredential A JSON string representing a CryptkeeperVerifiableCredential.
+ * @returns A CryptkeeperVerifiableCredential object. Throws error if the object is not a valid CryptkeeperVerifiableCredential.
+ */
+export async function deserializeCryptkeeperVerifiableCredential(
+  serializedCryptkeeperVerifiableCredential: string,
+): Promise<CryptkeeperVerifiableCredential> {
+  if (!serializedCryptkeeperVerifiableCredential) {
+    throw new Error("Serialized Cryptkeeper Verifiable Credential is not provided!");
+  }
+
+  let parsedCryptkeeperVerifiableCredential;
+  try {
+    parsedCryptkeeperVerifiableCredential = JSON.parse(serializedCryptkeeperVerifiableCredential) as {
+      verifiableCredential: string;
+      metadata: VerifiableCredentialMetadata;
+    };
+  } catch (error) {
+    throw new Error(`Serialized Cryptkeeper Verifiable Credential is not valid JSON: ${(error as Error).message}`);
+  }
+
+  return {
+    verifiableCredential: await deserializeVerifiableCredential(
+      parsedCryptkeeperVerifiableCredential.verifiableCredential,
+    ),
+    metadata: parsedCryptkeeperVerifiableCredential.metadata,
+  };
+}
+
+/**
+ * Serializes a VerifiableCredential object into a JSON string.
+ * @param verifiableCredential An object representing a VerifiableCredential.
+ * @returns A string representing a VerifiableCredential.
+ * @throws Error if the VerifiableCredential is not provided.
+ */
+export function serializeVerifiableCredential(verifiableCredential: VerifiableCredential): string {
+  if (!verifiableCredential) {
+    throw new Error("Verifiable Credential is not provided!");
+  }
+
+  return stringify(verifiableCredential);
+}
+
+/**
+ * Attempts to parse a JSON string into a VerifiableCredential object.
+ * @param serializedVerifiableCredential A JSON string representing a VerifiableCredential.
+ * @returns An object representing a VerifiableCredential. Throws error if the object is not a valid VerifiableCredential.
+ */
+export async function deserializeVerifiableCredential(
+  serializedVerifiableCredential: string,
+): Promise<VerifiableCredential> {
+  if (!serializedVerifiableCredential) {
+    throw new Error("Serialized Verifiable Credential is not provided!");
+  }
+
+  let deserializedVerifiableCredential;
+  try {
+    deserializedVerifiableCredential = JSON.parse(serializedVerifiableCredential) as Record<string, unknown>;
+  } catch (error) {
+    throw new Error(`Serialized Verifiable Credential is not valid JSON: ${(error as Error).message}`);
+  }
+
+  return validateVerifiableCredential(deserializedVerifiableCredential);
+}
+
+let claimValueSchema: yup.Schema<ClaimValue>;
```

Comment:
You can get rid of let and use const with variable initialization.

Reply:
The use of let here addresses a circular dependency naming issue that comes when `claimValueSchema` is not declared before the body of `claimValueMapSchema`

Reply:
Ok, this one works for now. In the future we can use class to get rid of hoisting.